src/storage/log_page_async_fetcher.cpp

```diff
@@ -0,0 +1,60 @@
+#include "log_page_async_fetcher.hpp"
+
+#include "log_impl.h"
+#include "thread_manager.hpp"
+
+namespace cublog
+{
+  class log_page_fetch_task : public cubthread::entry_task
+  {
+    public:
+      explicit log_page_fetch_task (
+	      LOG_PAGEID pageid,
+	      async_page_fetcher::callback_func_type &&callback)
+	:
+	cubthread::entry_task (),
+	m_logpageid (pageid),
+	m_callback (std::move (callback))
+      {
+      }
+
+      void execute (context_type &) override
+      {
+	log_lsa loglsa;
+	loglsa.pageid = m_logpageid;
+
+	log_reader logreader;
+	int err = logreader.set_lsa_and_fetch_page (loglsa);
+
+	auto page = logreader.get_page ();
+
+	m_callback (page, err);
```

Comment:
Nitpicking: make the code more compact:
```suggestion
	log_lsa loglsa { m_logpageid, 0 };
	log_reader logreader;
	
	int err = logreader.set_lsa_and_fetch_page (loglsa);

	m_callback (logreader.get_page (), err);
```